### PR TITLE
Fix/balance checks

### PR DESCRIPTION
**Pull request by sendra**

### CODE REVIEW

File: src/contracts/StakedTokenV2Rev4.sol [hunk 1]
```diff
@@ -1527,6 +1527,8 @@ contract StakedTokenV2Rev4 is
 
       emit TokensRescued(tokens[i], aaveMerkleDistributor, amounts[i]);
     }
+
+    require(IERC20(address(this)).totalSupply() == STAKED_TOKEN.balanceOf(address(this)), 'DES_COLLATERALIZED_STK_AAVE');
```
Comment (eboadom) [inline src/contracts/StakedTokenV2Rev4.sol:1531]: Better to use the own totalSupply, no?

Comment (eboadom) [reply src/contracts/StakedTokenV2Rev4.sol:1531]: Also better error `INVALID_COLLATERALIZATION` error

File: src/contracts/LendToAaveMigrator.sol [hunk 2]
```diff
@@ -63,6 +63,11 @@ contract LendToAaveMigrator is VersionedInitializable {
 
         emit LendMigrated(address(this), lendAmount);
         emit AaveTokensRescued(address(this), aaveMerkleDistributor, amountToRescue);
+
+        // checks that the amount of Aave not migrated is less or equal than the amount Aave disposable for migration
```
Comment (eboadom) [inline src/contracts/LendToAaveMigrator.sol:67]: "Aave" -> "AAVE"

File: test/LendToAaveMigrator.t.sol [hunk 3]
```diff
@@ -14,7 +14,7 @@ contract LendToAaveMigratorTest is Test {
     IERC20 public constant LEND = IERC20(0x80fB784B7eD66730e8b1DBd9820aFD29931aab03);
     uint256 public constant LEND_AAVE_RATIO = 100;
 
-    uint256 public constant lendAmountToMigrate = 8007719287288096435418;
+    uint256 public constant lendAmountToMigrate = 8007719287288096435418 + 841600717506653731350931;
```
Comment (eboadom) [inline test/LendToAaveMigrator.t.sol:17]: Why the new addition?

Comment (sendra) [reply test/LendToAaveMigrator.t.sol:17]: we were not taking into account also the amount of lend to lend contract amount to reascue, which is used on the actual payload, so added it also on the test.

**Review by eboadom — APPROVED** [review]

**Pull request merged**